pandas/tests/indexing/test_loc.py

```diff
@@ -1095,3 +1096,42 @@ def test_loc_datetimelike_mismatched_dtypes():
 
     with pytest.raises(KeyError, match=msg):
         df["a"].loc[tdi]
+
+
+def test_loc_setitem_df_datetime64tz_column_with_index():
+    df = pd.DataFrame(
+        pd.date_range("2020-01-01", "2020-01-06", 6, tz=timezone.utc), columns=["data"]
+    )
+    df2 = pd.DataFrame(index=df.index)
+    df2.loc[df.index, "data"] = df["data"]
+    tm.assert_numpy_array_equal(np.array(df.data), np.array(df2.data))
```

Comment:
pls use assert_frame_equal for cmparision

Reply:
@jreback please take a look at threads - https://github.com/pandas-dev/pandas/pull/32479#discussion_r388695828 and https://github.com/pandas-dev/pandas/pull/32479#discussion_r389034552 

`assert_frame_equals` will only work if the implementation is along the lines of https://github.com/pandas-dev/pandas/pull/32479#discussion_r392056376 and https://github.com/pandas-dev/pandas/pull/32479#discussion_r395987552

I can make the change if there is consensus on an alternate implementation.